src/safeposix/syscalls/net_calls.rs

```diff
@@ -2861,6 +2896,7 @@ impl Cage {
             );
         }
 
+        // get the flags
         let nonblocking = (socktype & SOCK_NONBLOCK) != 0;
         let cloexec = (socktype & SOCK_CLOEXEC) != 0;
 
```

Comment:
below i see the following comment:

"// we need to increment the refcount of the sockets we created
  // reason: in bind_inner_socket, we added entries to the inode table"
  
  please add some details to the comment. specifically what is a refcount of the sockets. I think this would make it clearer as to why we need to increment it if we add those entries to the inode table